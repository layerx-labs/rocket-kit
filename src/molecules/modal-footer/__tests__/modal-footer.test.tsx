import React from 'react';
import { render, screen } from '@testing-library/react';
import ModalFooter from '..';
import { Button } from '../../..';
import userEvent from '@testing-library/user-event';

describe('Modal Footer', () => {
  it('renders', () => {
    const { asFragment } = render(
      <ModalFooter closeValue={'Cancel'}>
        <Button type="submit" value="save" />
      </ModalFooter>
    );
    expect(asFragment()).toMatchSnapshot();
  });

  it('call close action', () => {
    const mockAction = jest.fn();
    render(
      <ModalFooter closeAction={mockAction}>
        <Button type="submit" value="save" />
      </ModalFooter>
    );

    userEvent.click(screen.getByText(/close/i));
    expect(mockAction).toBeCalledTimes(1);
  });

  it('children button is visible and clickable', () => {
    const mockAction = jest.fn();
    render(
      <ModalFooter>
        <Button type="submit" value="save" action={mockAction} />
      </ModalFooter>
    );

    userEvent.click(screen.getByText(/save/i));
    expect(mockAction).toBeCalledTimes(1);
  });

  it('focus mode button is visible and clickable', () => {
    const mockAction = jest.fn();
    render(
      <ModalFooter
        focusMode
        focusModeAction={mockAction}
        focusModeValue="Focus mode on"
      >
        <Button type="submit" value="save" />
      </ModalFooter>
    );

    userEvent.click(screen.getByText(/focus mode on/i));
    expect(mockAction).toBeCalledTimes(1);
  });
});
