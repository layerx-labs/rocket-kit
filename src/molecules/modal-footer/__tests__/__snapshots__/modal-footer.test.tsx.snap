// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal Footer renders 1`] = `
<DocumentFragment>
  .c1 {
  --button: hsl(186,62%,59%);
  --txt: hsl(0,0%,100%);
  --hover: #2ba8b6;
  --button: hsl(0,0%,48%);
  --hover: #616161;
  border-width: 3px;
  border-style: solid;
  border-color: var(--button);
  border-radius: 999px;
  background-color: transparent;
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  height: 2.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1.25rem;
  text-transform: uppercase;
  white-space: nowrap;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  cursor: pointer;
}

.c1:hover {
  border-color: var(--hover);
  background-color: transparent;
}

.c1:hover span {
  color: var(--hover);
}

.c1:hover svg {
  fill: var(--hover);
}

.c1:disabled {
  cursor: inherit;
  opacity: 0.5;
}

.c1:disabled:hover {
  pointer-events: none;
}

.c1 > *:not(:last-child) {
  margin-right: 5px;
}

.c1 .spinner {
  border-top-color: var(--button);
}

.c1 span {
  position: relative;
  font-size: 0.75rem;
  font-weight: 700;
  color: var(--button);
  pointer-events: none;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c1 svg {
  width: auto;
  min-width: 1.25rem;
  height: 1.25rem;
  fill: var(--button);
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c2 {
  --button: hsl(186,62%,59%);
  --txt: hsl(0,0%,100%);
  --hover: #2ba8b6;
  --button: hsl(186,62%,59%);
  --hover: #2798a5;
  border-width: 0;
  border-style: solid;
  border-color: var(--button);
  border-radius: 999px;
  background-color: var(--button);
  width: -webkit-min-content;
  width: -moz-min-content;
  width: min-content;
  height: 2.25rem;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 0 1.25rem;
  text-transform: uppercase;
  white-space: nowrap;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
  cursor: pointer;
}

.c2:hover {
  background-color: var(--hover);
}

.c2:disabled {
  cursor: inherit;
  opacity: 0.5;
}

.c2:disabled:hover {
  pointer-events: none;
}

.c2 > *:not(:last-child) {
  margin-right: 5px;
}

.c2 span {
  position: relative;
  font-size: 0.75rem;
  font-weight: 700;
  color: var(--txt);
  pointer-events: none;
  -webkit-transition-duration: 0.3s;
  transition-duration: 0.3s;
}

.c2 svg {
  width: auto;
  min-width: 1.25rem;
  height: 1.25rem;
  fill: var(--txt);
  -webkit-transition: 0.3s;
  transition: 0.3s;
}

.c0 {
  position: relative;
  margin-top: 3.125rem;
  max-width: 100%;
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-box-pack: center;
  -webkit-justify-content: center;
  -ms-flex-pack: center;
  justify-content: center;
}

.c0 > button {
  min-width: 6.25rem;
}

.c0 > button:not(:last-child) {
  margin-right: 0.3125rem;
}

@media (min-width:1280px) {
  .c1 {
    min-width: 6.25rem;
  }
}

@media (min-width:1280px) {
  .c2 {
    min-width: 6.25rem;
  }
}

@media (min-width:700px) {
  .c0 {
    -webkit-box-pack: end;
    -webkit-justify-content: flex-end;
    -ms-flex-pack: end;
    justify-content: flex-end;
  }
}

<div
    class="c0 modal-footer"
  >
    <button
      class="c1 button"
      color="info"
      value="Cancel"
    >
      <span>
        Cancel
      </span>
    </button>
    <button
      class="c2 button"
      color="primary"
      type="submit"
      value="save"
    >
      <span>
        save
      </span>
    </button>
  </div>
</DocumentFragment>
`;
